### DISCUSSION TITLE
support scripts using module

### DISCUSSION THREAD

**Discussion by quinn** [gnat/surreal#54] [Ideas]
from mdn docs:

https://developer.mozilla.org/en-US/docs/Web/API/Document/currentScript
> The Document.currentScript property returns the [<script>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script) element whose script is currently being processed and [isn't a JavaScript module](https://github.com/whatwg/html/issues/997). (For modules use [import.meta](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import.meta) instead.)

i'm basically doing:

```
<script type="module">
	import { thing } from './thing.js';
	thing(me());
</script>
```

**Comment by quinn** [comment]
Ok, import.meta isn't helpful here. It looks like there is an open issue on the spec for this:

https://github.com/whatwg/html/issues/1013

**Comment by gnat** [comment]
Thanks for looking into this. Would love to make it happen once it's in available.
